mne/coreg.py

```diff
@@ -1245,3 +1248,656 @@ def _scale_xfm(subject_to, xfm_fname, mri_name, subject_from, scale,
+            Relative weight for RPA. The default value is 1.
+        %(verbose)s
+
+        Returns
+        -------
+        self : Coregistration
+            The modified Coregistration object.
+        """
+        if verbose:
+            self._log_dig_mri_distance()
+        if self._scale_mode is None:
+            n_scale_params = 0
+        elif self._scale_mode == 'uniform':
+            n_scale_params = 1
+        else:
+            n_scale_params = 3
+        self._lpa_weight = lpa_weight
+        self._nasion_weight = nasion_weight
+        self._rpa_weight = rpa_weight
+
+        # Initial guess (current state)
+        est = self._parameters
+        est = est[:[6, 7, None, 9][n_scale_params]]
+
+        # Do the fits, assigning and evaluating at each step
+        for _ in range(n_iterations):
+            head_pts, mri_pts, weights = self._setup_icp(n_scale_params)
+            est = fit_matched_points(mri_pts, head_pts, scale=n_scale_params,
+                                     x0=est, out='params', weights=weights)
+            if n_scale_params == 0:
+                self._update_params(rot=est[:3], tra=est[3:6])
+            elif n_scale_params == 1:
+                est = np.array(list(est) + [est[-1]] * 2)
+                self._update_params(rot=est[:3], tra=est[3:6], sca=est[6:9])
+            else:
+                self._update_params(rot=est[:3], tra=est[3:6], sca=est[6:9])
+            angle, move, scale = self._changes
+            if angle <= self._icp_angle and move <= self._icp_distance and \
+                    all(scale <= self._icp_scale):
+                break
+        if verbose:
+            self._log_dig_mri_distance()
+        return self
+
+    def omit_head_shape_points(self, distance):
+        """Exclude head shape points that are far away from the MRI head.
+
+        Parameters
+        ----------
+        distance : float
+            Exclude all points that are further away from the MRI head than
+            this distance (in m.). A value of distance <= 0 excludes nothing.
+
+        Returns
+        -------
+        self : Coregistration
+            The modified Coregistration object.
+        """
+        distance = float(distance)
+        if distance <= 0:
+            return
+
+        # find the new filter
+        mask = self._orig_hsp_point_distance <= distance
+        n_excluded = np.sum(~mask)
+        logger.info("Coregistration: Excluding %i head shape points with "
+                    "distance >= %.3f m.", n_excluded, distance)
+        # set the filter
+        self._extra_points_filter = mask
+        self._update_params(force_update_omitted=True)
+        return self
+
+    def compute_dig_head_distances(self):
+        """Compute distance between head shape points and MRI skin surface.
+
+        Returns
+        -------
+        dist : array, shape (n_points,)
+            The distance of the head shape points to the MRI skin surface.
+        """
```

Comment:
Why not use `dig_mri_distances` here? Will refactor.

Reply:
Ohh I see why, because it's going to be slower to use the `surface` one. Hmm... I'll get rid of the uses of `dig_mri_distances` then